autogen.sh

```diff
@@ -14,9 +14,16 @@
 # ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
 # OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 
+set -eu
 cd `dirname "$0"`
 
-if [ "$LIBTOOLIZE" = "" ] && [ "`uname`" = "Darwin" ]; then
+if [ "${1:-dev}" == "release" ]; then
+    export LIBUV_RELEASE=true
+else
+    export LIBUV_RELEASE=false
```

Comment:
```suggestion
    export LIBUV_RELEASE=
```
Otherwise it _always_ overwrites m4/libuv-check-versions.m4, won't it?

Reply:
It only overwrites it after the `libuv-check-versions.m4` runs and passes. But this is not an if-set check, but is actually treating `false` as a program name to execute (which returns 1 causing the if block to be skipped)